spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java

```diff
@@ -133,32 +137,15 @@ else if (this.flywayDataSource != null) {
 			else {
 				flyway.setDataSource(this.dataSource);
 			}
+			if (!CollectionUtils.isEmpty(this.flywayCallbacks)) {
+				for (FlywayCallback flywayCallback : this.flywayCallbacks) {
+					flyway.setCallbacks(flywayCallback);
```

Comment:
This logic means that only the last callback will be used. A test that verifies that multiple callbacks are called, and that they're called in a well-defined order would be useful.